src/Catty/DataModel/Configuration/PenConfiguration.swift

```diff
@@ -24,17 +24,28 @@ struct PenConfiguration {
     var penDown = false
     static let sizeConversionFactor = 0.634
 
-    private(set) var size: CGFloat = SpriteKitDefines.defaultCatrobatPenSize * CGFloat(PenConfiguration.sizeConversionFactor)
+    private(set) var size: CGFloat
+    let screenRatio: CGFloat
 
     var catrobatSize: CGFloat {
         set {
-            size = CGFloat(PenConfiguration.sizeConversionFactor) * newValue
+            size = CGFloat(PenConfiguration.sizeConversionFactor) * newValue * screenRatio
         }
         get {
-            size / CGFloat(PenConfiguration.sizeConversionFactor)
+            (size / CGFloat(PenConfiguration.sizeConversionFactor)) / screenRatio
         }
     }
 
     var color = SpriteKitDefines.defaultPenColor
     var previousPosition: CGPoint?
+
+    init(projectHeaderScreenWidth width: CGFloat, projectHeaderScreenHeight height: CGFloat) {
```

Comment:
Please change the name to `projectWidth` and `projectHeight`. Suggestion: in case width and height are not available, you are handing over the screen dimensions from `CBSpriteNode`. This is a great idea, however when doing so, the `screenRatio` will always be 1 -- the identity element of the multiplication and division so that is fine. I would suggest making `projectWidth` and `projectHeight` optional and set the `screenRatio` directly in those cases (this can make the code in `CBSpriteNode` more readable and prevents calling the initializer in three code execution paths). Another advantage is that we can test this case directly in `PenConfigurationTests`. What do you think?

Reply:
This is a really good idea. Thanks!